tests/cases/system_test.py

```diff
@@ -434,3 +456,144 @@ def testLogRoute(self):
             '=== Last 0 bytes of %s/info.log: ===\n\n' % logRoot)
 
         del config.getConfig()['logging']
+
+    def testAccessFlags(self):
```

Comment:
can you add the following test cases?

 * acl_mixin hasAccessFlags
 * acl_mixin requireAccessFlags
 * setPublicFlags with a non-admin user where force=True
 * setAccessList with a non-admin user where force=True
 * setUserAccess with a non-admin user where force=True
 * setGroupAccess with a non-admin user where force=True
 * setPublicFlags passing in a single flag instead of a list/set/tuple
 * an admin-only access flag that is set by the admin on a group, and then set by a non-admin (the flag should still remain because it already exists on the group, you tested this case for user but not for group)
 * model_base hasAccessFlags called with flags=None